core/chain_makers.go

```diff
@@ -105,6 +105,10 @@ func (b *BlockGen) AddTx(tx *types.Transaction) {
 	b.receipts = append(b.receipts, receipt)
 }
 
+func (b *BlockGen) AddReceipt(receipt *types.Receipt) {
+	b.receipts = append(b.receipts, receipt)
+}
```

Comment:
I'll rename this `AddUncheckedReceipts` or suggest some other better name. This method I have already use on several occasions now and serves its purpose perfectly for those specific tests.